scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,309 @@
+    uri = f"https://api.firecloud.org/api/workspaces/{project}/{workspace_name}?fields=owners,workspace.createdBy,workspace.authorizationDomain"
+
+    # Get access token and and add to headers for requests.
+    # -H  "accept: application/json" -H  "Authorization: Bearer [token]
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json"}
+
+    # capture response from API and parse out status code
+    response = requests.get(uri, headers=headers)
+    status_code = response.status_code
+
+    if status_code == 404:          # if workspace does not exist
+        return False, None, None
+    elif status_code == 200:        # if workspace already exists
+        return True, status_code, response
+    else:                           # if other error
+        return None, None, response.json()
+
+
+def create_workspace(workspace_name, auth_domain_name, project="anvil-datastorage"):
+    """Create the Terra workspace with given authorization domain."""
+
+    # check if workspace already exists
+    ws_exists, ws_exists_status_code, ws_exists_response = check_workspace_exists(workspace_name, project)
+
+    if not ws_exists:  # if workspace doesn't exist, create workspace
+        # create request JSON
+        create_ws_json = make_create_workspace_request(workspace_name, auth_domain_name, project)  # json for API request
+
+        # request URL for createWorkspace
+        uri = f"https://api.firecloud.org/api/workspaces"
+
+        # Get access token and and add to headers for requests.
+        # -H  "accept: application/json" -H  "Authorization: Bearer [token] -H  "Content-Type: application/json"
+        headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json", "Content-Type": "application/json"}
+
+        # capture response from API and parse out status code
+        response = requests.post(uri, headers=headers, data=json.dumps(create_ws_json))
+        status_code = response.status_code
+
+        # print success or fail message based on status code
+        if status_code != 201:
+            print(f"WARNING: Failed to create workspace with name: {workspace_name}. Check output file for error details.")
+            return response.json()
+        else:
+            print(f"Successfully created workspace with name: {workspace_name}.")
+            return status_code
+    elif ws_exists:  # if workspace already exists
+        print(f"Workspace already exists with name: {project}/{workspace_name}.")
+        print(f"Existing workspace details: {json.dumps(ws_exists_response.json(), indent=2)}")
+        # make user decide if they want to update/overwrite existing workspace
+        while True:  # try until user inputs valid response
+            update_existing_ws = input("Would you like to continue modifying the existing workspace? (Y/N)" + "\n")
+            if update_existing_ws.upper() in ["Y", "N"]:
+                break
```

Comment:
what's this doing?

Reply:
if someone puts in a value thats not an n or y (like an int or some other random letter by accident) itll keep prompting them until they put in something valid.

Reply:
ah got it. might be helpful to add a comment to say that, like `break  # don't need to keep prompting` just to help our future selves :)